### PR TITLE
 seek: Allow seek buffers to resize their underlying data slices

### PR DESCRIPTION
**Pull request by ny0m**

### CODE REVIEW

File: seek.go [hunk 1]
```diff
@@ -74,12 +73,25 @@ func (s *SeekBuffer) Rest(cursor int) []byte {
...
 }
 
+// Resize removes redundant packets from the buffer by replacing the buffer's data
+// with a new array that is the remainder of the buffer after the cursor.
+func (s *SeekBuffer) Resize() {
+	s.dataLock.Lock()
+	defer s.dataLock.Unlock()
+
+	data := make([][]byte, len(s.data)-s.Cursor())
+	copy(data, s.data[s.Cursor():])
```
Comment (carlaKC) [inline seek.go]: Assumption that `len(s.data)>=s.Cursor()` always holds, yes? 

Not sure on the correctness of this, but benchmark in [2] is pretty interesting. 
[1] https://github.com/go101/go101/wiki/How-to-perfectly-clone-a-slice%3F
[2] https://github.com/go101/go101/wiki/How-to-efficiently-clone-a-slice%3F

File: seek.go [hunk 2]
```diff
@@ -7,13 +7,12 @@ import (
 const errSeekBeforeBuffer Error = "cannot seek past the start of the buffer"
 const errSeekBeyondBuffer Error = "cannot seek past the end of the buffer"
 
-// NewSeekBuffer returns a buffer that allows streaming from any point in its saved data.
-// SeekBuffers are thread-safe.
```
Comment (carlaKC) [inline seek.go:11]: Why are these comments gone?

**Review by carlaKC — APPROVED** [review]
Like the test coverage for the new function 🐬 
Just a nit + comment about how `Resize` reallocs to a new slice, but def an over optimization (and maybe not even true).

File: seek.go (hunk 2 above)
Comment (ny0m) [reply seek.go:11]: Moved some of them to document the actual struct

**Pull request merged**